src/CachePlugin.php

```diff
@@ -66,8 +67,12 @@ final class CachePlugin implements Plugin
      *              Defaults to an empty array
      * }
      */
-    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
+    public function __construct(CacheItemPoolInterface $pool, $streamFactory, array $config = [])
     {
+        if (!($streamFactory instanceof StreamFactory) && !($streamFactory instanceof StreamFactoryInterface)) {
+            throw new \LogicException(\sprintf('StreamFactory must be an instance of %s or %s', StreamFactory::class, StreamFactoryInterface::class));
```

Comment:
Why don't we throw a `TypeError` here?

Reply:
oh good idea. @deguif could you change this to throw a TypeError instead?

Reply:
Yes I can change this ;)

Reply:
And ✅